src/t/d11.clj

```diff
@@ -0,0 +1,88 @@
+(ns t.d11
+  (:gen-class)
+  (:require
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]
+    [clojure.core.match :refer [match]]))
+
+(def sample
+"125 17")
+
+(defn parse-input
+  [input]
+  (->> input
+       (re-seq #"\d+")
+       (map parse-long)
+  ))
+
+(defn split-stone-2
+  [stone]
+  (let [str-stone (str stone)
+        mid-i (quot (count str-stone) 2)]
+    (->>
+      [(take mid-i str-stone) (drop mid-i str-stone)]
+      (map (fn [l] (apply str l)))
+      (map parse-long))))
+
+(defn evolve-stone
+  [stone]
+  (cond
+   (= stone 0) [1]
+   (even? (count (str stone))) (split-stone-2 stone)
+   :else [(* stone 2024)]))
+
+(def memo-evolve-stone (memoize evolve-stone))
+
+(defn d11p1
+  [input]
+  (let [x (parse-input input)
+        stones-start (map (fn [s] [s 0]) x)
+        ]
+    (loop [stones stones-start
+           count-final-stones 0]
+      (if (empty? stones) count-final-stones
+        (let [[[stone n-blinks] & stones] stones
+              n-stones (memo-evolve-stone stone)]
+          (if (= 25 (inc n-blinks))
+            (recur  stones (+ count-final-stones (count n-stones)))
+            (recur (concat (map (fn [s] [s (inc n-blinks)]) n-stones) stones) count-final-stones)))))))
+
+(defn d11p2
+  [input]
+  (let [x (parse-input input)
+        stones-start (->> x
+                          (map (fn [s] {s 1}))
+                          (reduce (fn [n el] (merge-with + n el)))
+                          )
+        ]
```

Comment:
I think this is equivalent to
```suggestion
  (let [stones-start (frequencies (parse-input input))]
```

Reply:
(which at that point you'd probably put directly into the opening of the `loop` rather than have a separate `let`, I guess.)

Reply:
yes, definitely
But I ended up needing something to merge my maps later so I re-used this at that time, so not totally wasted